### PR TITLE
feat: add ruby server example

### LINKED ISSUE #96: Add an example of Ruby dev env and base image.
We should have an example to easily onboard Ruby developers. 

After this issue, we can prompt user to ask for common Ruby dependencies. 
https://github.com/buildsafedev/bsf/issues/89

### PR DESCRIPTION
**Pull request by hanshal101**
Fixes https://github.com/buildsafedev/bsf/issues/96
Closes https://github.com/buildsafedev/bsf/issues/96

# Description
This PR adds the Ruby Server Example for buildsafe.

# Note
Since buildsafe currently does not officially ruby, there is an alternate way to build and push the application, i.e
```shell
# to develop
bsf develop

# to push with tags (optional)
bsf oci pkgs --tag=dev --push
```

### CODE REVIEW

**Comment by hanshal101** [comment]
Hey @dr-housemd I have made the changes, please do review it!

File: ruby-server-example/README.md
Comment (rakshitgondwal) [inline ruby-server-example/README.md:1]: We can have a bit more detailed README, like entering into the development shell and showing how we can perform certain Ruby actions, then creating a base image with Ruby dependencies and using it the Dockerfile.
Basically, this guide should be self sufficient for a Ruby developer to know how bsf can help him/her.

File: ruby-server-example/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Ruby Alpine image (after using bsf dockerfile digest)
+FROM index.docker.io/library/ruby:3.1-alpine@sha256:f9c75043ce289b61cebdd0a16851d5ae68d82bfb912eb7d6ca4f57cc6780ecef AS build
```
Comment (rakshitgondwal) [inline ruby-server-example/Dockerfile]: We need to show the use of our own base image over here, not the docker one. So in the guide when we show how base image is created, we can show the use of that over here.

File: ruby-server-example/bsf.hcl [hunk 2]
```diff
@@ -0,0 +1,16 @@
+
+packages {
+  development = ["coreutils-full@9.5", "bash@5.2.15"]
```
Comment (rakshitgondwal) [inline ruby-server-example/bsf.hcl]: Rust development dependencies need to be added here which can later be shown in the development shell.

Comment (hanshal101) [reply ruby-server-example/bsf.hcl]: you mean ruby*

**Comment by dr-housemd** [comment]
@hanshal101 can you please only commit changes for Ruby and not modify other example folders? Lot of noise to review

File: ruby-server-example/bsf.hcl [hunk 3]
```diff
@@ -0,0 +1,16 @@
+
+packages {
+  development = ["coreutils-full@9.5", "bash@5.2.15", "ruby@~3.3.4"]
```
Comment (dr-housemd) [inline ruby-server-example/bsf.hcl:3]: Ruby's package manager bundle should also be present here.

**Comment by hanshal101** [comment]
> @hanshal101 can you please only commit changes for Ruby and not modify other example folders? Lot of noise to review

Getting few errors with git inside wsl, CONTINUE https://github.com/buildsafedev/examples/pull/4

**Pull request closed without merging**